pkg/cel/conversions.go

```diff
@@ -45,6 +45,15 @@ func GoNativeType(v ref.Val) (interface{}, error) {
 		return v.ConvertToNative(reflect.TypeOf([]interface{}{}))
 	case types.MapType:
 		return v.ConvertToNative(reflect.TypeOf(map[string]interface{}{}))
+	case types.OptionalType:
+		opt := v.(*types.Optional)
+		if opt.HasValue() {
+			return GoNativeType(opt.GetValue())
+		}
+		// this looks weird, but it is the easiest way to get any generic value
+		// that conforms to interface
+		var i interface{}
+		return &i, nil
```

Comment:
It looks like `GoNativeType` is quite similar to `v.Value()` (this addition is similar to https://github.com/google/cel-go/blob/8890f56dd657d3f4746ad1c53f55b65574457d29/common/types/optional.go#L114)

Can `GoNativeType` leverage or be replaced by `v.Value()` ?

Reply:
Loosely related-

I realize that his conversion code is to get values OUT of CEL.  But I wanted to mention that  Kubernetes has `UnstructuredToVal` and `TypedToVal` utilities under https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/apiserver/pkg/cel/common to make it easier to get Kuberentes flavored objects IN to CEL.

Reply:
I will check on if we can replace the value method. On first glance it looks it could be reused but there are some special cases such as `Null`:

```
// Value implements ref.Val.Value.
func (n Null) Value() any {
	return structpb.NullValue_NULL_VALUE
}
```

this looks very different to how we handle it. Not saying it couldnt be used, but that would likely lead to a much larger changeset. Open to if we want to change this here.

Reply:
On a side note, is there any way we can return a "any type" in kubernetes JSON Schema? Ideally i would like to return a nil value that resolves to an empty json scheme property. but this gets rejected with

```
CustomResourceDefinition.apiextensions.k8s.io "deploymentservices.kro.run" is invalid: spec.validation.openAPIV3Schema.properties[status].properties[optionalUnavailable].type: Required value: must not be empty for specified object fields
```

I would like to be able to do this because for optional values, KRO / we are not able to determine a correct type so it must be a catch all definition that can then be used in the CRD

Reply:
I changed to v.Value() now for this instance as I believe I made it backwards compatible, but there are other usages of GoNativeType that I didnt touch. @a-hilaly what is our stance on GoNativeType in general?